mailchimp/mailchimp-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mailChimp/MailChimpAuthenticator.java

```diff
@@ -0,0 +1,306 @@
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            String callbackUrl = getCallbackUrl(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            OAuthClientRequest accessRequest =
+                    getAccessRequest(tokenEndPoint, clientId, code, clientSecret, callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            if (StringUtils.isBlank(accessToken)) {
+                throw new AuthenticationFailedException("Access token is empty or null");
+            }
+            context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+            Map<ClaimMapping, String> claims = getSubjectAttributes(oAuthResponse, authenticatorProperties);
+            String email = claims.get(ClaimMapping.build(MailChimpAuthenticatorConstants.MailChimp_EMAIL, MailChimpAuthenticatorConstants.MailChimp_EMAIL, (String) null, false));
+            AuthenticatedUser authenticatedUserObj =
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(email);
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(email);
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Authentication process failed", e);
+        }
+    }
+
+    @Override
+    protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        HashMap claims = new HashMap();
+
+        try {
+            String e = token.getParam("access_token");
+            String url = this.getUserInfoEndpoint(token, authenticatorProperties);
+            String json = sendRequest(url, e);
+            JSONObject obj = new JSONObject(json);
+            if (StringUtils.isBlank(json)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to fetch user claims. Proceeding without user claims");
+                }
+
+                return claims;
+            }
+
+            Map jsonObject = JSONUtils.parseJSON(json);
+            Iterator i$ = jsonObject.entrySet().iterator();
+
+            while (i$.hasNext()) {
+                Map.Entry data = (Map.Entry) i$.next();
+                String key = (String) data.getKey();
+                claims.put(ClaimMapping.build(key, key, (String) null, false), jsonObject.get(key).toString());
+                if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable("UserClaims")) {
```

Comment:
Fixed in https://github.com/wso2-attic/is-connectors/pull/69/commits/09b8bec061544ea378d2b948e680c0a594a76b5b